In [266]:
import pandas as pd

In [267]:
begInv = pd.read_csv("BegInvFINAL12-31-16Sample.csv")

In [268]:
begInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  onHand  Price   startDate  
0  750mL       8  12.99  2016-01-01  
1  750mL       7  10.99  2016-01-01

In [269]:
begInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  
0  750mL               8  12.99  2016-01-01  
1  750mL               7  10.99  2016-01-01

In [270]:
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8  12.99  2016-01-01             103.92  
1  750mL               7  10.99  2016-01-01              76.93

In [271]:
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.head(2)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
1               49917          838079.59
2               52925          840452.20

# Deliverable 1 a

In [272]:
begInv_Store.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852         3291170.24
73             162551         3142497.36
67             158996         3079578.63
66             149314         2973033.90
76             140208         2952418.44
69             144255         2946726.65
38             114368         2232698.77
55             119641         2001263.66
50              94720         1649808.22
79              95330         1503149.48

# Deliverable 1 b

In [273]:
begInv_Brand = begInv[['Brand','01_01_2016_Qty','InvCost01_01_2016']].groupby('Brand').sum()
begInv_Brand.head(2)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
58                281            3650.19
60                288            3165.12

In [274]:
begInv_Brand.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
3545            14499          463823.01
1233            12016          432455.84
8068            15341          383371.59
4261            15499          340823.01
3858            13649          327439.51
2753             4625          286703.75
8082             9287          278517.13
8680             7066          275503.34
2589             6766          270572.34
3876            14829          266773.71

# Deliverable 2 a

In [275]:
endInv = pd.read_csv("EndInvFINAL12-31-16Sample.csv")

In [276]:
endInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  onHand  Price     endDate  
0  750mL      11  12.99  2016-12-31  
1  750mL       7  36.99  2016-12-31

In [277]:
endInv = endInv.rename(columns = {'onHand':'12_31_2016_Qty'})
endInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  
0  750mL              11  12.99  2016-12-31  
1  750mL               7  36.99  2016-12-31

In [278]:
endInv['InvCost12_31_2016'] = (endInv['12_31_2016_Qty'] * endInv['Price'])
endInv.head(2)

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11  12.99  2016-12-31             142.89  
1  750mL               7  36.99  2016-12-31             258.93

In [279]:
endInv_Store = endInv[['Store','InvCost12_31_2016','12_31_2016_Qty']].groupby('Store').sum()

In [280]:
endInv_Store.head(2)

InvCost12_31_2016  12_31_2016_Qty
Store                                   
1             1206845.93           79827
2              850884.06           56671

In [281]:
endInv_Store.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

InvCost12_31_2016  12_31_2016_Qty
Store                                   
50            4887260.68          260717
73            3254662.81          164589
67            3076114.82          163765
34            3074616.75          145829
76            2975945.18          143866
69            2968678.82          150848
66            2860504.99          144579
74            2803645.13          166015
38            2463906.85          129397
55            2234836.35          125584

# Deliverable 2b

In [282]:
endInv_Brand = endInv[['Brand','12_31_2016_Qty','InvCost12_31_2016']].groupby('Brand').sum()
endInv_Brand.head(2)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
58                385            5001.15
60                146            1604.54

In [283]:
endInv_Brand.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
1233            15047          526494.53
3545            16770          502932.30
2753             7849          470861.51
8068            15608          366631.92
3405            12268          355649.32
4261            16769          351981.31
2757            11603          336370.97
2589             7922          300956.78
1376            13180          276648.20
2585            10487          272557.13

# Deliverable 3

In [284]:
begInv.head()

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   
2  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   
3  1_HARDERSFIELD_63      1  HARDERSFIELD     63   Herradura Reposado Tequila   
4  1_HARDERSFIELD_72      1  HARDERSFIELD     72         No. 3 London Dry Gin   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8  12.99  2016-01-01             103.92  
1  750mL               7  10.99  2016-01-01              76.93  
2  750mL               6  36.99  2016-01-01             221.94  
3  750mL               3  38.99  2016-01-01             116.97  
4  750mL               6  34.99  2016-01-01             209.94

In [285]:
endInv.head()

InventoryId  Store          City  Brand                  Description   
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake  \
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   
2  1_HARDERSFIELD_63      1  HARDERSFIELD     63   Herradura Reposado Tequila   
3  1_HARDERSFIELD_72      1  HARDERSFIELD     72         No. 3 London Dry Gin   
4  1_HARDERSFIELD_75      1  HARDERSFIELD     75    Three Olives Tomato Vodka   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11  12.99  2016-12-31             142.89  
1  750mL               7  36.99  2016-12-31             258.93  
2  750mL               7  38.99  2016-12-31             272.93  
3  750mL               4  34.99  2016-12-31             139.96  
4  750mL               7  14.99  2016-12-31             104.93

In [286]:
merge3 = pd.merge(left=begInv, right=endInv, how='outer', on=['InventoryId'])
merge3.head()

InventoryId  Store_x        City_x  Brand_x   
0  1_HARDERSFIELD_58      1.0  HARDERSFIELD     58.0  \
1  1_HARDERSFIELD_60      1.0  HARDERSFIELD     60.0   
2  1_HARDERSFIELD_62      1.0  HARDERSFIELD     62.0   
3  1_HARDERSFIELD_63      1.0  HARDERSFIELD     63.0   
4  1_HARDERSFIELD_72      1.0  HARDERSFIELD     72.0   

                 Description_x Size_x  01_01_2016_Qty  Price_x   startDate   
0  Gekkeikan Black & Gold Sake  750mL             8.0    12.99  2016-01-01  \
1       Canadian Club 1858 VAP  750mL             7.0    10.99  2016-01-01   
2     Herradura Silver Tequila  750mL             6.0    36.99  2016-01-01   
3   Herradura Reposado Tequila  750mL             3.0    38.99  2016-01-01   
4         No. 3 London Dry Gin  750mL             6.0    34.99  2016-01-01   

   InvCost01_01_2016  Store_y        City_y  Brand_y   
0             103.92      1.0  HARDERSFIELD     58.0  \
1              76.93      NaN           NaN      NaN   
2             221.94      1.0  HARDERSFIELD     62.0   
3             116.97      1.0  HARDERSFIELD     63.0   
4             209.94      1.0  HARDERSFIELD     72.0   

                 Description_y Size_y  12_31_2016_Qty  Price_y     endDate   
0  Gekkeikan Black & Gold Sake  750mL            11.0    12.99  2016-12-31  \
1                          NaN    NaN             NaN      NaN         NaN   
2     Herradura Silver Tequila  750mL             7.0    36.99  2016-12-31   
3   Herradura Reposado Tequila  750mL             7.0    38.99  2016-12-31   
4         No. 3 London Dry Gin  750mL             4.0    34.99  2016-12-31   

   InvCost12_31_2016  
0             142.89  
1                NaN  
2             258.93  
3             272.93  
4             139.96

In [287]:
merge3 = merge3[['InventoryId', '01_01_2016_Qty', 'InvCost01_01_2016', '12_31_2016_Qty', 'InvCost12_31_2016',]]
merge3.head()

InventoryId  01_01_2016_Qty  InvCost01_01_2016  12_31_2016_Qty   
0  1_HARDERSFIELD_58             8.0             103.92            11.0  \
1  1_HARDERSFIELD_60             7.0              76.93             NaN   
2  1_HARDERSFIELD_62             6.0             221.94             7.0   
3  1_HARDERSFIELD_63             3.0             116.97             7.0   
4  1_HARDERSFIELD_72             6.0             209.94             4.0   

   InvCost12_31_2016  
0             142.89  
1                NaN  
2             258.93  
3             272.93  
4             139.96

# Deliverable 4 - Using the Purchases Data

## Deliverable 4a

In [288]:
purchases = pd.read_csv("PurchasesFINAL12-31-16Sample.csv")
purchases.head(2)

InventoryId  Store  Brand                   Description   Size   
0  69_MOUNTMEND_8412     69   8412     Tequila Ocho Plata Fresno  750mL  \
1  34_PITMERDEN_5215     34   5215  TGI Fridays Long Island Iced  1.75L   

   VendorNumber                   VendorName  PONumber      PODate   
0           105  ALTAMAR BRANDS LLC               8124  2015-12-21  \
1          4466  AMERICAN VINTAGE BEVERAGE        8137  2015-12-22   

  ReceivingDate InvoiceDate     PayDate  PurchasePrice  Quantity  Dollars   
0    2016-01-02  2016-01-04  2016-02-16          35.71         6   214.26  \
1    2016-01-02  2016-01-07  2016-02-21           9.41         5    47.05   

   Classification  
0               1  
1               1

In [289]:
purchases2 = purchases[["VendorName", "Dollars"]].groupby(["VendorName"]).sum().sort_values(by='Dollars' , ascending=False)
purchases2.head(10)

Dollars
VendorName                              
DIAGEO NORTH AMERICA INC     21315299.75
MARTIGNETTI COMPANIES        11709222.19
JIM BEAM BRANDS COMPANY      10232521.65
PERNOD RICARD USA            10224899.93
BACARDI USA INC               7420630.63
CONSTELLATION BRANDS INC      6584032.72
BROWN-FORMAN CORP             5719375.62
ULTRA BEVERAGE COMPANY LLP    5545766.29
E & J GALLO WINERY            5177067.61
M S WALKER INC                4647908.17

## Deliverable 4b

In [290]:
invoices = pd.read_csv("InvoicePurchases12-31-16Sample.csv")
invoices.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber   
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124  \
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6   214.26     3.47      NaN  
1  2015-12-22  2016-02-21        15   140.55     8.57      NaN

In [291]:
invoices2 = invoices[["VendorName", "Freight"]].groupby(["VendorName"]).sum().sort_values(by='Freight' , ascending=False)
invoices2.head(10)

Freight
VendorName                            
DIAGEO NORTH AMERICA INC     257032.07
MARTIGNETTI COMPANIES        144719.92
JIM BEAM BRANDS COMPANY      123880.97
PERNOD RICARD USA            123780.22
BACARDI USA INC               89286.27
CONSTELLATION BRANDS INC      79528.99
BROWN-FORMAN CORP             68601.68
ULTRA BEVERAGE COMPANY LLP    68054.70
E & J GALLO WINERY            61966.91
M S WALKER INC                55551.82

## Deliverable 4c PRACTICE

In [292]:
freight_purchases=pd.merge(left=purchases2,right=invoices2,on='VendorName',how='outer')
freight_purchases.head(2)

Dollars    Freight
VendorName                                         
DIAGEO NORTH AMERICA INC     21315299.75  257032.07
MARTIGNETTI COMPANIES        11709222.19  144719.92

In [293]:
at_least= freight_purchases[freight_purchases['Dollars']>=250000]
at_least.head()

Dollars    Freight
VendorName                                         
DIAGEO NORTH AMERICA INC     21315299.75  257032.07
MARTIGNETTI COMPANIES        11709222.19  144719.92
JIM BEAM BRANDS COMPANY      10232521.65  123880.97
PERNOD RICARD USA            10224899.93  123780.22
BACARDI USA INC               7420630.63   89286.27

In [294]:
highest= freight_purchases.groupby('VendorName').sum()
highest.head(2)

Dollars  Freight
VendorName                                    
AAPER ALCOHOL & CHEMICAL CO    105.07     0.48
ADAMBA IMPORTS INTL INC      30175.00   367.52

In [295]:
highest.sort_values(by='VendorName',ascending=False).head(10)

Dollars   Freight
VendorName                                       
ZORVINO VINEYARDS              37795.32    448.68
WINE GROUP INC               2214435.68  27100.41
WILLIAM GRANT & SONS INC     2501169.32  30234.42
WESTERN SPIRITS BEVERAGE CO   151838.07   1933.19
WEIN BAUER INC                 15284.53    222.23
WALPOLE MTN VIEW WINERY         4545.83     46.74
VRANKEN AMERICA                82189.04   1048.97
VINILANDIA USA                236914.09   2802.64
VINEYARD BRANDS LLC             3844.63     54.23
VINEYARD BRANDS INC           501899.47   6015.86

In [296]:
lowest= freight_purchases.groupby('VendorName').sum()
lowest.head(2)

Dollars  Freight
VendorName                                    
AAPER ALCOHOL & CHEMICAL CO    105.07     0.48
ADAMBA IMPORTS INTL INC      30175.00   367.52

In [297]:
lowest.sort_values(by='VendorName',ascending=True).head(5)

Dollars  Freight
VendorName                                        
AAPER ALCOHOL & CHEMICAL CO        105.07     0.48
ADAMBA IMPORTS INTL INC          30175.00   367.52
ALISA CARR BEVERAGES             16935.18   172.00
ALTAMAR BRANDS LLC                4388.99    62.39
AMERICAN SPIRITS EXCHANGE          764.52     6.19

# Deliverable 4c - ACTUAL

In [298]:
grouped_invoices = invoices.groupby(['VendorNumber', 'VendorName']).agg({'Freight': 'sum', 'Dollars': 'sum'}).reset_index()

In [299]:
filtered_invoices = grouped_invoices[grouped_invoices['Dollars'] >= 250000]

In [300]:
filtered_invoices.loc[:, 'Freight_per_$'] = filtered_invoices['Freight'] / filtered_invoices['Dollars']

C:\Users\medic\AppData\Local\Temp\ipykernel_29648\1087969894.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_invoices.loc[:, 'Freight_per_$'] = filtered_invoices['Freight'] / filtered_invoices['Dollars']


In [301]:
sorted_invoices = filtered_invoices.sort_values(by='Freight_per_$', ascending=False)

In [302]:
print(sorted_invoices)

     VendorNumber                   VendorName    Freight      Dollars   
88           9625  WESTERN SPIRITS BEVERAGE CO    1933.19    361249.21  \
21           1590  DIAGEO CHATEAU ESTATE WINES    7259.75   1365472.83   
89           9744  FREDERICK WILDMAN & SONS       3999.93    759449.24   
9             653      STATE WINE & SPIRITS       8014.98   1529682.04   
99          17031  FLAG HILL WINERY & VINEYARD    1573.31    300403.20   
8             516  BANFI PRODUCTS CORP            8510.41   1628866.68   
44           4425        MARTIGNETTI COMPANIES  144719.92  27821473.91   
78           8673  STE MICHELLE WINE ESTATES     15919.70   3086650.70   
91           9815  WINE GROUP INC                27100.41   5258636.79   
83           9165  ULTRA BEVERAGE COMPANY LLP    68054.70  13210613.93   
80           8892  TRINCHERO FAMILY ESTATES      11595.97   2252211.31   
68           7245  PROXIMO SPIRITS INC.          38994.78   7588029.90   
84           9206  PHILLIPS PRODUCTS C

In [303]:
print(sorted_invoices.head(10))

    VendorNumber                   VendorName    Freight      Dollars   
88          9625  WESTERN SPIRITS BEVERAGE CO    1933.19    361249.21  \
21          1590  DIAGEO CHATEAU ESTATE WINES    7259.75   1365472.83   
89          9744  FREDERICK WILDMAN & SONS       3999.93    759449.24   
9            653      STATE WINE & SPIRITS       8014.98   1529682.04   
99         17031  FLAG HILL WINERY & VINEYARD    1573.31    300403.20   
8            516  BANFI PRODUCTS CORP            8510.41   1628866.68   
44          4425        MARTIGNETTI COMPANIES  144719.92  27821473.91   
78          8673  STE MICHELLE WINE ESTATES     15919.70   3086650.70   
91          9815  WINE GROUP INC                27100.41   5258636.79   
83          9165  ULTRA BEVERAGE COMPANY LLP    68054.70  13210613.93   

    Freight_per_$  
88       0.005351  
21       0.005317  
89       0.005267  
9        0.005240  
99       0.005237  
8        0.005225  
44       0.005202  
78       0.005158  
91       0.00515

In [304]:
print(sorted_invoices.tail(5))

    VendorNumber                   VendorName   Freight     Dollars   
75          8352  LUXCO INC                    10261.60  2051436.01  \
39          3924  HEAVEN HILL DISTILLERIES     14069.87  2816661.94   
35          3089  SIDNEY FRANK IMPORTING CO     8549.55  1715908.88   
92          9819  TREASURY WINE ESTATES        14836.57  2978686.40   
61          6359  OLE SMOKY DISTILLERY LLC      1922.00   387622.69   

    Freight_per_$  
75       0.005002  
39       0.004995  
35       0.004983  
92       0.004981  
61       0.004958  


# Deliverable 4d

In [305]:
filtered_invoices2 = invoices[(invoices['Freight'] > 100) & (invoices['Quantity'] <= 1000)]

In [306]:
filtered_invoices2_sorted = filtered_invoices2.sort_values(by=['Freight'], ascending=False)

In [307]:
print(filtered_invoices2_sorted)

      VendorNumber                   VendorName InvoiceDate  PONumber   
71            8664  STOLI GROUP,(USA) LLC        2016-01-12      8202  \
59            7239  REMY COINTREAU USA INC       2016-01-08      8163   
69             653      STATE WINE & SPIRITS     2016-01-08      8201   
50            6785  PALM BAY INTERNATIONAL INC   2016-01-08      8114   
407           2561  EDRINGTON AMERICAS           2016-02-01      8591   
23            2561  EDRINGTON AMERICAS           2016-01-08      8175   
40           10000  MAJESTIC FINE WINES          2016-01-09      8128   
28            3924  HEAVEN HILL DISTILLERIES     2016-01-08      8155   
38            8352  LUXCO INC                    2016-01-08      8113   
34            4692  KOBRAND CORPORATION          2016-01-11      8112   
11            1485  CASTLE BRANDS CORP.          2016-01-08      8152   
45            6213  MHW LTD                      2016-01-09      8144   
85            9625  WESTERN SPIRITS BEVERAGE CO  20

In [309]:
vendor_freight = filtered_invoices.groupby('VendorName')['Freight'].sum().sort_values("Freight", ascending=False)
vendor_freight.head()

TypeError: Series.sort_values() takes 1 positional argument but 2 positional arguments (and 1 keyword-only argument) were given

In [ ]:
max_freight_vendor = vendor_freight.idxmax() 
max_freight_value = vendor_freight.max()

In [ ]:
print(f"The vendor(s) with the highest freight cost under the given conditions is {max_freight_vendor} with a total freight cost of ${max_freight_value:.2f}.")

The vendor(s) with the highest freight cost under the given conditions is (3960, 'DIAGEO NORTH AMERICA INC   ') with a total freight cost of $257032.07.


In [ ]:
print(filtered_invoices2_sorted.head(5))

     VendorNumber                   VendorName InvoiceDate  PONumber   
71           8664  STOLI GROUP,(USA) LLC        2016-01-12      8202  \
59           7239  REMY COINTREAU USA INC       2016-01-08      8163   
69            653      STATE WINE & SPIRITS     2016-01-08      8201   
50           6785  PALM BAY INTERNATIONAL INC   2016-01-08      8114   
407          2561  EDRINGTON AMERICAS           2016-02-01      8591   

         PODate     PayDate  Quantity   Dollars  Freight Approval  
71   2015-12-25  2016-02-04       975  14299.98   349.02      NaN  
59   2015-12-23  2016-02-18       382   7515.38   348.07      NaN  
69   2015-12-25  2016-02-13       541   5544.89   303.70      NaN  
50   2015-12-20  2016-02-14       715   5533.18   242.73      NaN  
407  2016-01-22  2016-03-17       765  42239.46   219.65      NaN  


In [ ]:
columns_to_print = ['VendorNumber', 'VendorName', 'Freight', 'Quantity', 'Dollars']

In [ ]:
print(filtered_invoices2_sorted[columns_to_print].head(5))

     VendorNumber                   VendorName  Freight  Quantity   Dollars
71           8664  STOLI GROUP,(USA) LLC         349.02       975  14299.98
59           7239  REMY COINTREAU USA INC        348.07       382   7515.38
69            653      STATE WINE & SPIRITS      303.70       541   5544.89
50           6785  PALM BAY INTERNATIONAL INC    242.73       715   5533.18
407          2561  EDRINGTON AMERICAS            219.65       765  42239.46


# Deliverable 4e

In [ ]:
filtered_invoices2_sorted['Freight_per_$'] = filtered_invoices2_sorted['Freight'] / filtered_invoices2_sorted['Dollars']

In [ ]:
filtered_invoices2_sorted['Freight_per_unit'] = filtered_invoices2_sorted['Freight'] / filtered_invoices2_sorted['Quantity']

In [ ]:
print(filtered_invoices2_sorted)

      VendorNumber                   VendorName InvoiceDate  PONumber   
71            8664  STOLI GROUP,(USA) LLC        2016-01-12      8202  \
59            7239  REMY COINTREAU USA INC       2016-01-08      8163   
69             653      STATE WINE & SPIRITS     2016-01-08      8201   
50            6785  PALM BAY INTERNATIONAL INC   2016-01-08      8114   
407           2561  EDRINGTON AMERICAS           2016-02-01      8591   
23            2561  EDRINGTON AMERICAS           2016-01-08      8175   
40           10000  MAJESTIC FINE WINES          2016-01-09      8128   
28            3924  HEAVEN HILL DISTILLERIES     2016-01-08      8155   
38            8352  LUXCO INC                    2016-01-08      8113   
34            4692  KOBRAND CORPORATION          2016-01-11      8112   
11            1485  CASTLE BRANDS CORP.          2016-01-08      8152   
45            6213  MHW LTD                      2016-01-09      8144   
85            9625  WESTERN SPIRITS BEVERAGE CO  20

In [ ]:
columns_to_print1 = ['VendorName','Freight_per_$', 'Freight_per_unit']

In [ ]:
print(filtered_invoices2_sorted[columns_to_print1].head(5))

                      VendorName  Freight_per_$  Freight_per_unit
71   STOLI GROUP,(USA) LLC             0.024407          0.357969
59   REMY COINTREAU USA INC            0.046314          0.911178
69       STATE WINE & SPIRITS          0.054771          0.561368
50   PALM BAY INTERNATIONAL INC        0.043868          0.339483
407  EDRINGTON AMERICAS                0.005200          0.287124


In [ ]:
import statistics

In [ ]:
dataC= [.005002, .004995, .004983, .004981, .004958]

In [ ]:
average = statistics.mean(dataC)

In [ ]:
print("Average:", average)

Average: 0.0049838


In [ ]:
dataD = [.024407, .046314, .054771, .043868, .005200]

In [ ]:
averageD = statistics.mean(dataD)

In [ ]:
print("Average:", averageD)

Average: 0.034912


### How do they compare to findings in C above?

### The above analysis compares the average freight per dollar for the lists of vendors created in part C & D. The freight costs for 4C (large vendors) average at .0049838 and the freight costs for 4D average lower at 0.034912. This is helpful for this company to know if the want to reduce freight expenses. 

# Deliverable 5

In [ ]:
sales = pd.read_csv("salesDecember2016Sample.csv")
sales.head(5)

InventoryId  Store  Brand                   Description   Size   
0  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL  \
1  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   
2  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   
3  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   
4  10_HORNSEY_1003     10   1003  Crown Royal +2 Reusable Cups  750mL   

   SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume   
0              5        114.95       22.99  2016-12-01   750.0  \
1              4         91.96       22.99  2016-12-03   750.0   
2              2         45.98       22.99  2016-12-04   750.0   
3              1         22.99       22.99  2016-12-06   750.0   
4             11        252.89       22.99  2016-12-09   750.0   

   Classification  ExciseTax  VendorNo                VendorName  
0               1       3.94      3960  DIAGEO NORTH AMERICA INC  
1               1       3.15      3960  DIAGEO NORTH AMERICA INC  
2               1       1.57      3960  DIAGEO NORTH AMERICA INC  
3               1       0.79      3960  DIAGEO NORTH AMERICA INC  
4               1       8.66      3960  DIAGEO NORTH AMERICA INC

In [ ]:
sales1= sales.groupby('InventoryId').sum()
sales1.head(5)

Store   Brand   
InventoryId                       
10_HORNSEY_1003     110   11033  \
10_HORNSEY_10058     80   80464   
10_HORNSEY_10164     10   10164   
10_HORNSEY_1019      50    5095   
10_HORNSEY_10236    110  112596   

                                                        Description   
InventoryId                                                           
10_HORNSEY_1003   Crown Royal +2 Reusable CupsCrown Royal +2 Reu...  \
10_HORNSEY_10058  F Coppola Dmd Ivry Cab SvgnF Coppola Dmd Ivry ...   
10_HORNSEY_10164                         Andre Bourgogne Pnt Nr RSV   
10_HORNSEY_1019   Drambuie Glass PackDrambuie Glass PackDrambuie...   
10_HORNSEY_10236  Layer Cake Malbec MendozaLayer Cake Malbec Men...   

                                                               Size   
InventoryId                                                           
10_HORNSEY_1003   750mL750mL750mL750mL750mL750mL750mL750mL750mL7...  \
10_HORNSEY_10058           750mL750mL750mL750mL750mL750mL750mL750mL   
10_HORNSEY_10164                                              750mL   
10_HORNSEY_1019                           750mL750mL750mL750mL750mL   
10_HORNSEY_10236  750mL750mL750mL750mL750mL750mL750mL750mL750mL7...   

                  SalesQuantity  SalesDollars  SalesPrice   
InventoryId                                                 
10_HORNSEY_1003              79       1816.21      252.89  \
10_HORNSEY_10058             16        229.64      113.80   
10_HORNSEY_10164              1         15.99       15.99   
10_HORNSEY_1019               5        134.95      134.95   
10_HORNSEY_10236             21        250.95      131.45   

                                                          SalesDate  Volume   
InventoryId                                                                   
10_HORNSEY_1003   2016-12-012016-12-032016-12-042016-12-062016-1...  8250.0  \
10_HORNSEY_10058  2016-12-012016-12-022016-12-032016-12-132016-1...  6000.0   
10_HORNSEY_10164                                         2016-12-20   750.0   
10_HORNSEY_1019   2016-12-082016-12-102016-12-212016-12-232016-1...  3750.0   
10_HORNSEY_10236  2016-12-082016-12-092016-12-102016-12-162016-1...  8250.0   

                  Classification  ExciseTax  VendorNo   
InventoryId                                             
10_HORNSEY_1003               11      62.21     43560  \
10_HORNSEY_10058              16       1.78     16000   
10_HORNSEY_10164               2       0.11     10754   
10_HORNSEY_1019                5       3.95     18320   
10_HORNSEY_10236              22       2.34     48675   

                                                         VendorName  
InventoryId                                                          
10_HORNSEY_1003   DIAGEO NORTH AMERICA INCDIAGEO NORTH AMERICA I...  
10_HORNSEY_10058  SOUTHERN WINE & SPIRITS NESOUTHERN WINE & SPIR...  
10_HORNSEY_10164                                     PERFECTA WINES  
10_HORNSEY_1019   WILLIAM GRANT & SONS INCWILLIAM GRANT & SONS I...  
10_HORNSEY_10236  MARTIGNETTI COMPANIESMARTIGNETTI COMPANIESMART...

In [ ]:
sales1.sort_values(by='InventoryId',ascending=False).head(1)

Store  Brand   
InventoryId                     
9_BLACKPOOL_984     54   5904  \

                                                       Description   
InventoryId                                                          
9_BLACKPOOL_984  Patron XO Cafe LiqueurPatron XO Cafe LiqueurPa...  \

                                           Size  SalesQuantity  SalesDollars   
InventoryId                                                                    
9_BLACKPOOL_984  750mL750mL750mL750mL750mL750mL              9        233.91  \

                 SalesPrice   
InventoryId                   
9_BLACKPOOL_984      155.94  \

                                                         SalesDate  Volume   
InventoryId                                                                  
9_BLACKPOOL_984  2016-12-082016-12-132016-12-172016-12-212016-1...  4500.0  \

                 Classification  ExciseTax  VendorNo   
InventoryId                                            
9_BLACKPOOL_984               6       7.09     54990  \

                                                        VendorName  
InventoryId                                                         
9_BLACKPOOL_984  ULTRA BEVERAGE COMPANY LLPULTRA BEVERAGE COMPA...

In [ ]:
sales1 = (sales1['SalesPrice'] * sales1['SalesQuantity'])
sales1.head(1)

InventoryId
10_HORNSEY_1003    19978.31
dtype: float64

In [ ]:
purchases.head(2)

InventoryId  Store  Brand                   Description   Size   
0  69_MOUNTMEND_8412     69   8412     Tequila Ocho Plata Fresno  750mL  \
1  34_PITMERDEN_5215     34   5215  TGI Fridays Long Island Iced  1.75L   

   VendorNumber                   VendorName  PONumber      PODate   
0           105  ALTAMAR BRANDS LLC               8124  2015-12-21  \
1          4466  AMERICAN VINTAGE BEVERAGE        8137  2015-12-22   

  ReceivingDate InvoiceDate     PayDate  PurchasePrice  Quantity  Dollars   
0    2016-01-02  2016-01-04  2016-02-16          35.71         6   214.26  \
1    2016-01-02  2016-01-07  2016-02-21           9.41         5    47.05   

   Classification  
0               1  
1               1

In [ ]:
purchases1= purchases.groupby('InventoryId').sum()
purchases1.head(5)

Store  Brand   
InventoryId                      
10_HORNSEY_1001      20   2002  \
10_HORNSEY_1003      60   6018   
10_HORNSEY_10030     10  10030   
10_HORNSEY_10058     30  30174   
10_HORNSEY_10062     10  10062   

                                                        Description   
InventoryId                                                           
10_HORNSEY_1001            Bailey's 50mL 4 PackBailey's 50mL 4 Pack  \
10_HORNSEY_1003   Crown Royal +2 Reusable CupsCrown Royal +2 Reu...   
10_HORNSEY_10030                            Pacifico Sur Pnt Nr Rsv   
10_HORNSEY_10058  F Coppola Dmd Ivry Cab SvgnF Coppola Dmd Ivry ...   
10_HORNSEY_10062                                  Terroirs du Rhone   

                                            Size  VendorNumber   
InventoryId                                                      
10_HORNSEY_1001               50mL 4 Pk50mL 4 Pk          7920  \
10_HORNSEY_1003   750mL750mL750mL750mL750mL750mL         23760   
10_HORNSEY_10030                           750mL         90017   
10_HORNSEY_10058                 750mL750mL750mL          6000   
10_HORNSEY_10062                           750mL         10754   

                                                         VendorName  PONumber   
InventoryId                                                                     
10_HORNSEY_1001   DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC...     22780  \
10_HORNSEY_1003   DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC...     77870   
10_HORNSEY_10030                        THE IMPORTED GRAPE LLC           8614   
10_HORNSEY_10058  SOUTHERN WINE & SPIRITS NE SOUTHERN WINE & SPI...     35846   
10_HORNSEY_10062                        PERFECTA WINES                   8298   

                                                             PODate   
InventoryId                                                           
10_HORNSEY_1001                                2016-07-142016-08-10  \
10_HORNSEY_1003   2016-09-282016-10-132016-10-252016-11-162016-1...   
10_HORNSEY_10030                                         2016-01-24   
10_HORNSEY_10058                     2016-06-232016-09-202016-10-31   
10_HORNSEY_10062                                         2016-01-01   

                                                      ReceivingDate   
InventoryId                                                           
10_HORNSEY_1001                                2016-07-192016-08-16  \
10_HORNSEY_1003   2016-10-042016-10-192016-11-042016-11-252016-1...   
10_HORNSEY_10030                                         2016-02-05   
10_HORNSEY_10058                     2016-06-282016-09-292016-11-10   
10_HORNSEY_10062                                         2016-01-06   

                                                        InvoiceDate   
InventoryId                                                           
10_HORNSEY_1001                                2016-08-012016-08-30  \
10_HORNSEY_1003   2016-10-132016-10-312016-11-092016-12-032016-1...   
10_HORNSEY_10030                                         2016-02-16   
10_HORNSEY_10058                     2016-07-042016-10-082016-11-20   
10_HORNSEY_10062                                         2016-01-15   

                                                            PayDate   
InventoryId                                                           
10_HORNSEY_1001                                2016-09-072016-10-05  \
10_HORNSEY_1003   2016-11-242016-11-292016-12-222017-01-062017-0...   
10_HORNSEY_10030                                         2016-03-18   
10_HORNSEY_10058                     2016-08-202016-11-172016-12-18   
10_HORNSEY_10062                                         2016-02-16   

                  PurchasePrice  Quantity  Dollars  Classification  
InventoryId                                                         
10_HORNSEY_1001            8.56       120   513.60               2  
10_HORNSEY_1003           99.96        46   766.36 

In [ ]:
purchases1.sort_values(by='InventoryId',ascending=False).head(1)

Store  Brand   
InventoryId                     
9_BLACKPOOL_984     27   2952  \

                                                       Description   
InventoryId                                                          
9_BLACKPOOL_984  Patron XO Cafe LiqueurPatron XO Cafe LiqueurPa...  \

                            Size  VendorNumber   
InventoryId                                      
9_BLACKPOOL_984  750mL750mL750mL         27495  \

                                                        VendorName  PONumber   
InventoryId                                                                    
9_BLACKPOOL_984  ULTRA BEVERAGE COMPANY LLP ULTRA BEVERAGE COMP...     31970  \

                                         PODate   
InventoryId                                       
9_BLACKPOOL_984  2016-01-062016-04-282016-12-20  \

                                  ReceivingDate   
InventoryId                                       
9_BLACKPOOL_984  2016-01-142016-05-042016-12-28  \

                                    InvoiceDate   
InventoryId                                       
9_BLACKPOOL_984  2016-01-192016-05-172017-01-03  \

                                        PayDate  PurchasePrice  Quantity   
InventoryId                                                                
9_BLACKPOOL_984  2016-02-242016-06-122017-02-02          61.89        13  \

                 Dollars  Classification  
InventoryId                               
9_BLACKPOOL_984   268.19               3

In [ ]:
purchases1 = (purchases1['PurchasePrice'] * purchases1['Quantity'])
purchases1.head(1)

InventoryId
10_HORNSEY_1001    1027.2
dtype: float64

### The above two tables show the top product being purchased and sold. This product is the 9_BLACKPOOL_984

In [ ]:
data = {
    'product': ['1003', '10058', '10164', '1019'],
    'revenue': [1816.21, 229.64, 15.99, 134.95],
    'variable_costs': [8.56, 99.96, 5.19, 27.78]
}

In [ ]:
df = pd.DataFrame(data)

In [ ]:
df['contribution_margin'] = df['revenue'] - df['variable_costs']

In [ ]:
print(df)

  product  revenue  variable_costs  contribution_margin
0    1003  1816.21            8.56              1807.65
1   10058   229.64           99.96               129.68
2   10164    15.99            5.19                10.80
3    1019   134.95           27.78               107.17


### This analysis shows the products that are sold and purchased the most as well as calculates the contribution margin for these products.